### PR TITLE
General Cleanup

### PR DESCRIPTION
**Pull request by shankarshastri**
fixes: #122

### CODE REVIEW

File: src/main/scala/zio/zmx/MetricsDataModel.scala [hunk 1]
```diff
@@ -103,10 +103,10 @@ trait MetricsDataModel {
     sealed case class Timer(name: String, value: Double, sampleRate: Double, tags: Chunk[Label]) extends Metric[Double]
 
     case object Zero extends Metric[Int] {
-      def name      = ""
-      def tags      = Chunk.empty
-      def value     = 0
-      def isEmpty() = true
+      val name    = ""
```
Comment (alesavin) [inline src/main/scala/zio/zmx/MetricsDataModel.scala:106]: why here you prefer vals? I have some expectations here will be races in init and some of vals can be null, but maybe i'm wrong 
+ why not just

```suggestion
    val Zero = new Metric[Int] {
      def name      = ""
      def tags      = Chunk.empty
      def value     = 0
    }
```

File: src/main/scala/zio/zmx/diagnostics/ZMXConfig.scala [hunk 2]
```diff
@@ -16,8 +16,8 @@
 
 package zio.zmx.diagnostics
 
-case class ZMXConfig(host: String, port: Int, debug: Boolean)
+final case class ZMXConfig(host: String, port: Int, debug: Boolean)
 
 object ZMXConfig {
-  def empty = new ZMXConfig(host = "localhost", port = 1111, debug = false)
+  val empty = ZMXConfig(host = "localhost", port = 1111, debug = false)
```
Comment (alesavin) [inline src/main/scala/zio/zmx/diagnostics/ZMXConfig.scala:22]: val Empty (with capital start letter) if we follow https://docs.scala-lang.org/style/naming-conventions.html#constants-values-variable-and-methods

**Review by adamgfraser — APPROVED** [review]
Great cleanup!

**Comment by adamgfraser** [comment]
@shankarshastri Can you resolve conflicts and then we can get this merged?

File: src/main/scala/zio/zmx/MetricsDataModel.scala (hunk 1 above)
Comment (shankarshastri) [reply src/main/scala/zio/zmx/MetricsDataModel.scala:106]: @alesavin , I prefer to keep it val as there is no need for construction of each of the attributes for Metric using methods.

**Comment by shankarshastri** [comment]
@adamgfraser , Merge conflict fixed :)

**Pull request merged**